addons/stock/models/stock_quant.py

```diff
@@ -189,7 +189,7 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
 
     def write(self, vals):
         """ Override to handle the "inventory mode" and create the inventory move. """
-        if self._is_inventory_mode() and 'inventory_quantity' in vals:
+        if self._is_inventory_mode() and 'inventory_quantity' in vals or 'removal_date' in vals:
```

Comment:
removal date is defined in product_expiry, you can't use it here
you'd need to remove the "and 'inventory_quantity' in vals " part and instead replace it by a check looking it there is any keys from _get_inventory_fields_write present in the vals